site/app/libraries/database/DatabaseQueries.php

```diff
@@ -1100,18 +1110,46 @@ public function getParentPostId($child_id) {
         return $this->course_db->rows()[0]['parent_id'];
     }
 
-    public function editPost($original_creator, $user, $post_id, $content, $anon, $markdown) {
+    /**
+     * @return string[]
+     */
+    public function getForumAttachments(int $post_id, int $version = 0) {
```

Comment:
Oh, the temptation to use this function in a loop....

It would be better to accept an array of post IDs, and then return the results as an associative array with those post IDs as the keys.